### PR TITLE
Adding in option to filter projects that DynamicPlatformResolution was unable to resolve

### PR DESCRIPTION
**Pull request by MIchaelRShea**
…

Fixes #

### Context
Currently dynamic platform resolution will leave the "platform" global property of a depending project blank if it is unable to determine the correct architecture to build a project as. This is helpful but in certain situations, especially traversal style projects it makes more sense to remove dependent projects that don't build as a compatible platform.

### Changes Made
I added a property that can be added to projects so that if dynamic platform resolution is unable to determine what platform a project should be built as it will be removed from the graph.

### Notes

linking bug https://github.com/dotnet/msbuild/issues/8890

### CODE REVIEW

**Comment by JanKrivanek** [comment]
@MIchaelRShea Thank you for your contribution.
Can you crerate/link a bug describing the issue? This would help properly triaging and processing this PR.

**Comment by MIchaelRShea** [comment]
> @MIchaelRShea Thank you for your contribution. Can you crerate/link a bug describing the issue? This would help properly triaging and processing this PR.

Yes went ahead and linked a bug to this PR

File: src/Build/Graph/ProjectInterpretation.cs [hunk 1]
```diff
@@ -35,6 +35,7 @@ internal sealed class ProjectInterpretation
         private const string PlatformMetadataName = "Platform";
         private const string PlatformsMetadataName = "Platforms";
         private const string EnableDynamicPlatformResolutionPropertyName = "EnableDynamicPlatformResolution";
+        private const string EnableDynamicPlatformFilteringPropertyName = "EnableDynamicPlatformFiltering";
```
Comment (JanKrivanek) [inline src/Build/Graph/ProjectInterpretation.cs:38]: This should be mentioned in https://github.com/dotnet/msbuild/blob/main/documentation/ProjectReference-Protocol.md

File: src/Build.UnitTests/Graph/GetCompatiblePlatformGraph_Tests.cs [hunk 2]
```diff
@@ -325,6 +325,37 @@ public void FailsToResolve()
...
+                                                    <Project>
+                                                    <PropertyGroup>
+                                                        <Platforms>x64</Platforms>
+                                                    </PropertyGroup>
+                                                    </Project>");
+
+                ProjectGraph graph = new ProjectGraph(entryProject.Path);
+                // Here we are checking if platform is defined. in this case it should not be since Platorm would be set to the value this project defaults as
+                // in order to avoid dual build errors we remove platform in order to avoid the edge case where a project has global platform set and does not have global platform set
+                // yet still default to the same platform.
```
Comment (JanKrivanek) [inline src/Build.UnitTests/Graph/GetCompatiblePlatformGraph_Tests.cs:354]: I might be just slow - but this comment is very hard to understand. Especially the `where a project has global platform set and does not have global platform set` feels wrong.
Can it be restated?

File: src/Build.UnitTests/Graph/GetCompatiblePlatformGraph_Tests.cs [hunk 3]
```diff
@@ -325,6 +325,37 @@ public void FailsToResolve()
             }
         }
 
+        [Fact]
```
Comment (JanKrivanek) [inline src/Build.UnitTests/Graph/GetCompatiblePlatformGraph_Tests.cs:328]: Turning this into theory with positive and negative case (2.proj set for matching and mismatching platform, then the expected number of nodes should be 1 and 0 respectively)

File: src/Build/Graph/ProjectInterpretation.cs [hunk 4]
```diff
@@ -179,13 +181,17 @@ public IEnumerable<ReferenceInfo> GetReferences(ProjectInstance requesterInstanc
 
                     var selectedPlatform = PlatformNegotiation.GetNearestPlatform(overridePlatformNegotiationMetadataValue, projectInstance.GetPropertyValue(PlatformMetadataName), projectInstance.GetPropertyValue(PlatformsMetadataName), projectInstance.GetPropertyValue(PlatformLookupTableMetadataName), requesterInstance.GetPropertyValue(PlatformLookupTableMetadataName), projectInstance.FullPath, requesterInstance.GetPropertyValue(PlatformMetadataName));
 
-                    if (selectedPlatform.Equals(String.Empty))
+                    if (enableDynamicPlatformFiltering && !selectedPlatform.Item1)
+                    {
+                        continue;
```
Comment (JanKrivanek) [inline src/Build/Graph/ProjectInterpretation.cs:186]: Project being removed from graph might deserve low-priority log message for clarity during troubleshooting.

**Review by JanKrivanek — CHANGES_REQUESTED** [review]
Overall looks good.

I just wish for higher clarity for future maintanance of the code - especially by adding documentation of the added property and it's expected behavior

**Pull request closed without merging**